src/components/Upload/UploadFile.vue

```diff
@@ -34,35 +56,58 @@ import axios from 'axios';
 export default {
 	data() {
 		return {
-			file: undefined,
+			dragAndDropCapable: false,
+            uploadPercentage: 0
 		};
 	},
+	mounted(){
+		this.dragAndDropCapable = this.determineDragAndDropCapable();              
+		
+		// If drag and drop capable, then we continue to bind events to our elements.
+		if ( this.dragAndDropCapable ){
+			
+			// Listen to all of the drag events and bind an event listener to each for the fileform.
+			['drag', 'dragstart', 'dragend', 'dragover', 'dragenter', 'dragleave', 'drop'].forEach( function( evt ) {
+			
+			// For each event add an event listener that prevents the default action (opening the file in the browser) 
+			// and stop the propagation of the event (so no other elements open the file in the browser)		
+			this.$refs.fileform.addEventListener(evt, function(e){
+				e.preventDefault();
+				e.stopPropagation();
+			}.bind(this), false);
+			}.bind(this));
+
+			// Add an event listener for drop to the form
+			this.$refs.fileform.addEventListener('drop', function(e){
+				//  Capture the files from the drop event and add them to our local files array.
+				for( let i = 0; i < e.dataTransfer.files.length; i++ ){
+					this.$parent.files.push( e.dataTransfer.files[i] );
+					console.log(this.$parent.files);                
+				}				
+			}.bind(this));	
+		}
+    }, 
 	methods: {
+		
+		determineDragAndDropCapable() {
+            var div = document.createElement('div');
+            return ( ( 'draggable' in div )
+                || ( 'ondragstart' in div && 'ondrop' in div ) )
+                && 'FormData' in window
+                && 'FileReader' in window;
+		},
+		
+		removeFile(key) {
+            this.$parent.files.splice( key, 1 );
+		},
+		
 		handleFileUpload(event) {
 			console.log(event);
-			// this.file = this.$refs.file.files;
-			this.file = this.$refs.file.file[0];
-		},
-		submitFile() {
-			let formData = new FormData();
-			/*for( var i = 0; i < this.files.length; i++ ){
-                let file = this.files[i];
+			//this.files = this.$refs.files.files[0];
 
-                formData.append('files[' + i + ']', file);
-            }*/
-			formData.append('file', this.file);
-			axios
-				.post('http://localhost:8000/file/upload', formData, {
-					headers: {
-						'Content-Type': 'multipart/form-data',
-					},
-				})
-				.then((res) => {
-					console.log(res);
-				})
-				.catch(function (e) {
-					console.log('Chyba', e);
-				});
+			for( let i = 0; i < this.$refs.files.files.length; i++ ){	
+				this.$parent.files.push( this.$refs.files.files[i]);            
```

Comment:
Again, please event emitting. And also, it looks a little similar to the one in the drag method above. Could we extract the funcitonality into separate method if it is needed?